packages/cli/generators/app/templates/src/application.ts

```diff
@@ -3,26 +3,38 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-import {Application, ApplicationConfig} from '@loopback/core';
-import {RestComponent, RestServer} from '@loopback/rest';
+import {Application, ApplicationConfig<% if (project.loopbackBoot) { %>, BootOptions<% } %>} from '@loopback/core';
+import {RestApplication, RestServer} from '@loopback/rest';
+<% if (project.loopbackBoot) { -%>
+import {BootComponent} from '@loopback/boot';
+import {Context} from '@loopback/context';
+<% } else {-%>
 import {PingController} from './controllers/ping.controller';
+<% } -%>
 import {MySequence} from './sequence';
 
-export class <%= project.applicationName %> extends Application {
+export class <%= project.applicationName %> extends RestApplication {
   constructor(options?: ApplicationConfig) {
     // Allow options to replace the defined components array, if desired.
-    options = Object.assign(
-      {},
-      {
-        components: [RestComponent],
-      },
-      options,
-    );
     super(options);
-    this.server(RestServer);
+<% if (project.loopbackBoot) { -%>
```

Comment:
IMO, we should leverage `BootComponent` in all projects scaffolded by our CLI tooling. (Remove `loopbackBoot` prompt, assume the answer was yes, and remove code like `setupControllers`.)

As I am envisioning our CLI tooling, it should give our users the recommended project setup. The more options we provide, the more difficult the CLI will be for novice users, and the more different project layouts we will have to support.